### DISCUSSION TITLE
Android melonDS cheats?

### DISCUSSION THREAD

**Discussion by 10nz1n** [rafaelvcaetano/melonDS-android#861] [Support]
Super Mario 64 DS was one of my favorite games growing up. Decided to download the rom and melonDS on my android phone. However, I don't really wanna grind out every single aspect of the game from scratch, so I wanted to use some cheats. However, I do not know how to add them to the app. I have some cheat codes, but no "xml" files. Help.

**Comment by rafaelvcaetano** [comment]
You can download this cheat database: https://bitbucket.org/DeadSkullzJr/nds-i-cheat-databases/src/master/Cheat%20Databases/cheats.xml

To download it on your device and assuming you're using Chrome, do the following:
1. Open the link
2. Click on the 3 horizontal dots next to "Edit"
3. Long press on "Open raw"
4. Click "Download link"

After the download completes, you can import the file and don't forget to enable the option "Enable cheats". You can then toggle individual cheats after launching the ROM, pressing Back to pause and clicking on Cheats.

**Reply by 10nz1n** [reply]
I did that and unfortunately there are 0 cheats for Super Mario 64 DS. Is there a way to convert cheat codes...
(example: Infinite health
1211117c 00000008
12092144 00000880)
...into a xml database or add them to this database?

**Reply by rafaelvcaetano** [reply]
I've just tested and I can find cheats for SM64 DS using the database I gave you. Are you using a modified ROM?

Also, there's currently no way to add cheats manually from inside the app but you should be able to edit the XMl file as you wish. You just need to follow the same format that is already used for other games. The most import part is to know the ROM's ID.

**Reply by 10nz1n** [reply]
0037 - Super Mario 64 DS (U)(Trashman).ml1
How do I edit the xml file?

**Reply by rafaelvcaetano** [reply]
You can do that using any file editor. You can download an app to do that on your device (just search for "text editor" or "code editor" in the Play Store) or you can do that on a PC (probably much easier) using the notepad.

**Reply by 10nz1n** [reply]
Alright, so basically I just google "cheat codes for Super Mario 64 DS" and edit the XML file to add those codes in?
I'll try it and see if it works.

**Reply by BigHeartDe13** [reply]
i have the English Patched version of Zoids Saga Legend of Arcadia. how would i add codes to the deadskulls database file that i already have? unless i cam find cheats for the japanese version that will work for it since it is just a patched version...
